src/full/Agda/TypeChecking/Monad/Base.hs

```diff
@@ -2960,7 +2960,7 @@ instance Pretty ProjLams where
 
 -- | Is the record type recursive?
 recRecursive :: Defn -> Bool
-recRecursive (Record { recMutual = Just qs }) = not $ null qs
+recRecursive (Record { recMutual = mqs }) = any (not . null) mqs
```

Comment:
Well, the semantics of `recMutual = Nothing` is that nothing is known about mutualilty yet.
If you want _not mutual_ you have to make sure this gets set to `Just []`.